### PR TITLE
OCPBUGS-25599: Reconcile AgentMachines for unbound Agents

### PR DESCRIPTION
**Pull request by avishayt**
AgentMachines need to be reconciled if they don't yet have an Agent, and and a valid Agent is modified. Otherwise AgentMachines that don't yet have an Agent will not be reconciled when an Agent becomes available.

### CODE REVIEW

File: controllers/agentmachine_controller.go [hunk 1]
```diff
@@ -675,31 +675,33 @@ func (r *AgentMachineReconciler) mapMachineToAgentMachine(ctx context.Context, m
...
+	if err := r.List(ctx, amList, opts); err != nil {
+		log.WithError(err).Error("failed to list agent machines")
+		return []reconcile.Request{}
+	}
 
-		reply := make([]reconcile.Request, 0, len(amList.Items))
+	reply := make([]reconcile.Request, 0, len(amList.Items))
+	if mappedAgent {
+		// If the Agent is mapped to an AgentMachine, return only that AgentMachine
+		agent := a
```
Comment (carbonin) [inline controllers/agentmachine_controller.go:704]: Why is this assignment needed?

**Review by CrystalChun — APPROVED** [review]
/lgtm

**Comment by CrystalChun** [comment]
/retest-required

Looks like a flake: 
```
[2023-12-18 15:35:31] Waiting for operator local-storage-operator to get installed on namespace openshift-local-storage...
[2023-12-18 15:35:31] Error from server (NotFound): subscriptions.operators.coreos.com "local-storage-operator" not found
```

**Pull request merged**